core/src/main/java/org/apache/iceberg/io/FanoutDataWriter.java

```diff
@@ -52,13 +51,7 @@ public FanoutDataWriter(FileWriterFactory<T> writerFactory, OutputFileFactory fi
 
   @Override
   protected FileWriter<T, DataWriteResult> newWriter(PartitionSpec spec, StructLike partition) {
-    // TODO: support ORC rolling writers
-    if (fileFormat == FileFormat.ORC) {
```

Comment:
When `ORC` support rolling writers, `fileFormat` is using in no where.
Should we deprecated/remove it?

Reply:
Since these methods involve multiple spark/flink versions, I suggest that a separate PR cleans up after this.